src/dlgPackageExporter.cpp

```diff
@@ -662,228 +665,231 @@ void dlgPackageExporter::slot_export_package()
-                    zip_close(archive);
-                    isOk = false;
-                }*/
+    // Process the config and the file containing the Mudlet triggers,
+    // etc. specially so they are inserted first and last respectively:
+    if (isOk) {
+        if (fileEntries.contains(QStringLiteral("config.lua"))) {
+            if (!writeFileToZip(QStringLiteral("config.lua"), fileEntries.value(QStringLiteral("config.lua")), archive).first) {
+                /* isOk = false; */
+            } else {
+                fileEntries.remove(QStringLiteral("config.lua"));
             }
+        }
+    }
 
-            QString xmlFileName = mPackageName;
-            xmlFileName.append(QLatin1String(".xml"));
-            if (isOk) {
-                QMapIterator<QString, QString> itFileName(fileEntries);
-                while (itFileName.hasNext() && isOk) {
-                    itFileName.next();
-                    if (itFileName.key() == xmlFileName) {
-                        continue;
-                    }
-
-                    if (!writeFileToZip(itFileName.key(), itFileName.value(), archive)) {
-                        zip_close(archive);
-                        isOk = false;
-                        break;
-                    }
-                }
+    QString xmlFileName = packageName;
+    xmlFileName.append(QLatin1String(".xml"));
+    if (isOk) {
+        QMapIterator<QString, QString> itFileName(fileEntries);
+        while (itFileName.hasNext() && isOk) {
+            itFileName.next();
+            if (itFileName.key() == xmlFileName) {
+                continue;
             }
 
-            if (isOk) {
-                if (fileEntries.contains(xmlFileName) && fileEntries.value(xmlFileName) == mXmlPathFileName) {
-                    if (!writeFileToZip(xmlFileName, mXmlPathFileName, archive)) {
-                        isOk = false;
-                    }
-
-                    // If successful will get to HERE...
-
-                } else {
-                    displayResultMessage(tr("Required file \"%1\" was not found in the staging area. "
-                                            "This area contains the Mudlet items chosen for the package, "
-                                            "which you selected to be included in the package file. "
-                                            "This suggests there may be a problem with that directory: "
-                                            "\"%2\" - "
-                                            "Do you have the necessary permissions and free disk-space?")
-                                            .arg(mXmlPathFileName, QDir(StagingDirName).canonicalPath()), false);
-                    isOk = false;
-                }
+            if (std::tie(isOk, error) = writeFileToZip(itFileName.key(), itFileName.value(), archive); !isOk) {
+                zip_close(archive);
```

Comment:
Not really handling this error properly unfortunately - this means we have not been able to stage the Mudlet items in the XML file ready to be written into the archive. However because we are using `zip_close(archive)` rather than `zip_discard(archive)` then anything **else** that has been (*successfully*) staged to go in will do so - but not this file. This may not be the best outcome...!